subsys/llext/llext_load.c

```
@@ -173,23 +173,33 @@ static int llext_find_tables(struct llext_loader *ldr, struct llext *ext)
 
 		switch (shdr->sh_type) {
 		case SHT_SYMTAB:
+			if (ldr->hdr.e_type == ET_REL) {
+				LOG_DBG("symtab at %d", i);
+				ldr->sects[LLEXT_MEM_SYMTAB] = *shdr;
+				ldr->sect_map[i].mem_idx = LLEXT_MEM_SYMTAB;
+				table_cnt++;
+			}
```

The missing else scenario seems suspicious to me, under what circumstances would the section type not imply an elf type here? Should this result in some sort of warning to a user that we don't support this particular elf in these alternate branch cases?

---

LLEXT already loads only `ET_REL` or `ET_DYN` files; other types get a meaningful rejection upfront [here](https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/llext/llext_load.c#L81-L93).
If there is a mismatch (say, only a dynamic section in a REL file), there is no `table_cnt++`, and there's already a check below that stops the load process with "Missing needed tables". 

The issue I wanted to prevent here is the opposite - finding _both_ tables in (debug-enabled) shared libraries. This way, only the dynamic symbol table is looked for in that loop, and the static one is silently ignored.

---

Ok I think I understand now, but then this switch would likely be better written as an if else with boolean expressions in my opinion. It looks a bit confusing to see a single if in the case statement here when really what I think we want is more like...

```c
if (elf_type == ET_REL && sect_type == SHT_SYMTAB) {
} else if (elf_type == ET_DYN && sect_type = SHT_DYNTAB) {
} else if (sect_type == SHT_STRTAB) {
}
```